src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,1050 @@
+
+        Not properly implemented as it was only used on Rabi amplitude vs lenght and it
+        was an unused routine.
+        """
+        self.zhsweepers = [self.select_sweeper(sweeper, qubit)]
+
+    def select_sweeper(self, sweeper, qubit):
+        """Sweeper translation"""
+
+        if sweeper.parameter is Parameter.amplitude:
+            zh_sweeper = lo.SweepParameter(
+                uid=sweeper.parameter.name,
+                values=sweeper.values,
+            )
+        elif sweeper.parameter is Parameter.duration:
+            zh_sweeper = lo.SweepParameter(
+                uid=sweeper.parameter.name,
+                values=sweeper.values * 1e-9,
+            )
+        # TODO: take intermediate frequency from the pulses
+        elif sweeper.parameter is Parameter.frequency:
+            if self.pulse.type is PulseType.READOUT:
+                intermediate_frequency = qubit.readout_frequency - qubit.readout.local_oscillator.frequency
+            elif self.pulse.type is PulseType.DRIVE:
+                intermediate_frequency = qubit.drive_frequency - qubit.drive.local_oscillator.frequency
+            zh_sweeper = lo.LinearSweepParameter(
+                uid=sweeper.parameter.name,
+                start=sweeper.values[0] + intermediate_frequency,
+                stop=sweeper.values[-1] + intermediate_frequency,
+                count=len(sweeper.values),
+            )
+        return zh_sweeper
+
+    def add_sweeper(self, sweeper, qubit):
+        """Add sweeper to list of sweepers"""
+        self.zhsweepers.append(self.select_sweeper(sweeper, qubit))
+
+
+class ZhSweeperLine:
+    """
+    Zurich sweeper from qibolab sweeper for non pulse parameters
+    Bias, Delay (, power_range, local_oscillator frequency, offset ???)
+
+    For now Parameter.bias sweepers are implemented as Parameter.Amplitude
+    on a flux pulse. We may want to keep this class separate for future
+    Near Time sweeps
+    """
+
+    def __init__(self, sweeper, qubit=None, sequence=None):
+        """Qibolab sweeper"""
+        self.sweeper = sweeper
+
+        # TODO: I already created a flux pulse check
+        if sweeper.parameter is Parameter.bias:
+            pulse = FluxPulse(
+                start=sequence.start,
+                duration=sequence.duration,
+                amplitude=1,
+                shape="Rectangular",
+                channel=qubit.flux.name,
+                qubit=qubit.name,
+            )
+            self.pulse = pulse
+            self.signal = f"flux{qubit.name}"
+
+            self.zhpulse = lo.pulse_library.const(
+                uid=(f"{pulse.type.name.lower()}_{pulse.qubit}_"),
+                length=round(pulse.duration * 1e-9, 9),
+                amplitude=pulse.amplitude,
+            )
+
+        # Need something better to store multiple sweeps on the same pulse
+        self.zhsweeper = self.select_sweeper(sweeper)
+
+    def select_sweeper(self, sweeper):
```

Comment:
`self` not used, but since I see a pattern, maybe you want the name not to be in the global scope:
```suggestion
    @staticmethod
    def select_sweeper(sweeper):
```